source/features/resolve-conflict.tsx

```diff
@@ -0,0 +1,186 @@
+import React from 'dom-chef';
+import CodeMirror from 'codemirror';
+import features from '../libs/features';
+
+interface Editor extends CodeMirror.Editor, CodeMirror.Doc {
+}
+interface EditorDOMElement extends HTMLDivElement {
+	CodeMirror: Editor;
+}
+
+function embeddedInit() {
+	class Resolver {
+		editor: Editor
+
+		conflictNumber = 1
+
+		constructor() {
+			const codeMirrorElement: EditorDOMElement = document.querySelector('.CodeMirror');
+			this.editor = codeMirrorElement.CodeMirror;
+
+			/**
+             * Listen to swapDoc event in order to find out file change
+             */
+			this.editor.on('swapDoc', () => {
+				// Add 200 milisecond delay, we need keep calm until codemirror updates its content
+				setTimeout(() => {
+					const conflicts = [...document.querySelectorAll('.CodeMirror .rg-conflict')];
+					if (conflicts.length === 0) {
+						this.processConflicts();
+					}
+				}, 200);
+			});
+		}
+
+		/**
+         * Process editor lines and find conflict lines, assign class to each of lines
+         * and add resolver widget to the first line of conflict
+         */
+		processConflicts() {
+			const lines = [...document.querySelectorAll('.CodeMirror .conflict-gutter-marker')];
+			let stepBranch = '';
+			lines.forEach(line => {
+				const lineNumber = parseInt(line.parentElement.parentElement.innerText, 10) - 1;
+				let cl = `rg-conflict c-${this.conflictNumber} `;
+				if (line.classList.contains('js-start')) {
+					cl += 'rg-separator rg-start incoming-branch current-branch';
+					stepBranch = 'incoming-branch';
+					this.addWidgetToLine(this.conflictNumber, lineNumber);
+				}
+
+				if (line.classList.contains('js-middle')) {
```

Comment:
The code inside appears to expect to be exclusive in each condition, let’s use `else if`